### PR TITLE
[Docs] General reformatting

### PR DESCRIPTION
**Pull request by DesyncTheThird**
## Description

General minor reformatting of documentation, from the Introduction page (newbs), up to and not including the PR Checklist (pr_checklist):
- Minor grammar/punctuation changes;
- Standardised indenting, use of inline code blocks, hyphenation, capitalisation, etc.;
- Fixed/removed various dead links;
- Removed some outdated information;
- Added more crosslinking.

Things I've noticed but don't know how to fix/am deferring to others' judgement before changing:
- The `Prev`/`Next pages` links for Bootmagic still say "Bootmagic Lite".
- On the Quantum Painter page, The `Previous page` link is missing, and the `Next page` link links to Introduction (newbs) instead of Quantum Painter LVGL Integration (quantum_painter_lvgl).
- The linked database of combo examples page (https://combos.gboards.ca/) at the end of Combo page doesn't render well for me; is there a different, more up-to-date page?
- There is some slight indenting after the code block in `faq_misc` that doesn't work due to the code block.
- The links to 2 EEPROM reset images at the end of `faq_build` are dead.
- Check that the table in `reference_configurator_support` renders correctly (currently, the json is not escaped and causes the page to not load.)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

I am struggling to get `qmk docs` working on my windows machine (have troubleshooted this briefly on the QMK discord), so I have not been able to test these changes. I may have access to a Linux machine in a week's time (w.c. 16th), so I will try test it then.

### CODE REVIEW

File: package-lock.json
Comment (daskygit) [inline package-lock.json:1]: Please remove this file.

File: .gitignore [hunk 1]
```diff
@@ -120,3 +120,9 @@ compile_commands.json
 via*.json
 /keyboards/**/keymaps/via/*
 /keyboards/**/keymaps/vial/*
+
+
+
+# ignore my keymap stuff
+/keyboards/crkbd/keymaps/
+/keyboards/sofle/keymaps/
```
Comment (daskygit) [inline .gitignore:128]:
```suggestion
```

Please revert this change.

**Review by daskygit — CHANGES_REQUESTED** [review]

**Comment by daskygit** [comment]
Ideally this could/should have been split into multiple PRs to make reviewing easier. I'm willing to read through as is but other reviews may not.

File: .gitignore (hunk 1 above)
Comment (DesyncTheThird) [reply .gitignore]: Does this commit work, or do I need to do something more specific to remove them properly.

**Comment by DesyncTheThird** [comment]
> Ideally this could/should have been split into multiple PRs to make reviewing easier. I'm willing to read through as is but other reviews may not.

Sorry, I wasn't sure if it was bad form or not to spam a bunch of small PRs, so I thought I'd just do one.

**Comment by zvecr** [comment]
Theres **far** too much change here to make it a reviewable PR. Especially as it contains some wider impacting style choices, like the switch to em dashes (which personally, I do not think adds any value and should be reverted).

File: .gitignore (hunk 1 above)
Comment (zvecr) [reply .gitignore]: If the file is still shown in the `Files changed` tab, then it still needs to be reverted.

**Comment by DesyncTheThird** [comment]
I can cut this up into smaller PRs then, before @daskygit spends any more time on this one.

About the em dashes, I noticed that they were being used in some pages and not others, so I figured it would be good to make them all the same, but I can revert those.

**Comment by zvecr** [comment]
> I can cut this up into smaller PRs

Please do this incrementally, in batches or something, where they are reviewed and merged before more are created. We would not want to receive 101 PRs, which potentially all suffer from the same paradigms.
